client/chain-spec/src/chain_spec.rs

```diff
@@ -378,6 +402,49 @@ where
 	fn set_storage(&mut self, storage: Storage) {
 		self.genesis = GenesisSource::Storage(storage);
 	}
+
+	fn set_hardcoded_sync(&mut self, hardcoded_sync: SerializableHardcodedSync) {
+		ChainSpec::set_hardcoded_sync(self, hardcoded_sync)
+	}
+}
+
+/// Hardcoded infomation that allows light clients to sync quickly.
+pub struct HardcodedSync<Block: BlockT> {
```

Comment:
This should be named different.

Maybe `LightSyncState` or similar to make more clear what this is.

Reply:
@danforbes any ideas? :P

Reply:
Basti's suggestion seems fine to me. I like that it implies up front that the struct is designed for light clients. Could also do something like `InjectableLightClientState`.